### PR TITLE
NavigationBar-Update (IOS-235)

### PR DESCRIPTION
**Pull request by zeitschlag**
Adds a menu to switch between different timelines. And it looks like this:

Simulator Screenshot - iPhone 15 Pro - 2024-04-01 at 20 12 09

And it works like this:

https://github.com/mastodon/mastodon-ios/assets/2580019/b7468285-4f23-41c2-b814-5612dbf159a0

### CODE REVIEW

**Comment by samhenrigold** [comment]
Looks really fantastic, nice work

File: Mastodon/Scene/HomeTimeline/HomeTimelineViewController.swift [hunk 1]
```diff
@@ -94,8 +169,10 @@ extension HomeTimelineViewController {
         settingBarButtonItem.target = self
         settingBarButtonItem.action = #selector(HomeTimelineViewController.settingBarButtonItemPressed(_:))
         
-        navigationItem.titleView = titleView
-        titleView.delegate = self
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(customView: timelineSelectorButton)
+
+//        navigationItem.titleView = titleView
```
Comment (DK1DA) [inline Mastodon/Scene/HomeTimeline/HomeTimelineViewController.swift:174]: We should get rid of this code

File: Mastodon/Scene/HomeTimeline/HomeTimelineViewModel+LoadLatestState.swift [hunk 2]
```diff
@@ -93,10 +114,20 @@ extension HomeTimelineViewModel.LoadLatestState {
...
             do {
                 await AuthenticationServiceProvider.shared.fetchAccounts(apiService: viewModel.context.apiService)
-                let response = try await viewModel.context.apiService.homeTimeline(
-                    authenticationBox: viewModel.authContext.mastodonAuthenticationBox
-                )
-                
+                let response: Mastodon.Response.Content<[Mastodon.Entity.Status]>
+
+                switch viewModel.timelineContext {
+                case .following:
```
Comment (DK1DA) [inline Mastodon/Scene/HomeTimeline/HomeTimelineViewModel+LoadLatestState.swift]: I'm somewhat unsure about the naming, maybe this should be called `home` and the other one `public` then it's easier to infer which of the timelines APIs is used.

File: MastodonSDK/Sources/MastodonSDK/MastodonFeed.swift [hunk 3]
```diff
@@ -6,9 +6,15 @@ import CoreDataStack
...
     
     public enum Kind {
-        case home
+
+        case home(timeline: HomeTimeline)
         case notificationAll
         case notificationMentions
+
+        public enum HomeTimeline {
+            case following
```
Comment (DK1DA) [inline MastodonSDK/Sources/MastodonSDK/MastodonFeed.swift]:
```suggestion
            case home
```

File: MastodonSDK/Sources/MastodonSDK/MastodonFeed.swift [hunk 4]
```diff
@@ -6,9 +6,15 @@ import CoreDataStack
...
+            case localCommunity
```
Comment (DK1DA) [inline MastodonSDK/Sources/MastodonSDK/MastodonFeed.swift]:
```suggestion
            case `public`
```

File: MastodonSDK/Sources/MastodonSDK/MastodonFeed.swift [hunk 5]
```diff
@@ -6,9 +6,15 @@ import CoreDataStack
 public final class MastodonFeed {
     
     public enum Kind {
-        case home
+
+        case home(timeline: HomeTimeline)
         case notificationAll
         case notificationMentions
+
+        public enum HomeTimeline {
```
Comment (DK1DA) [inline MastodonSDK/Sources/MastodonSDK/MastodonFeed.swift]:
```suggestion
        public enum TimelineContext {
```

**Review by kimar — APPROVED** [review]
LGTM, just a few questions and remarks but nothing blocking from merging.

File: Mastodon/Scene/HomeTimeline/HomeTimelineViewController.swift (hunk 1 above)
Comment (zeitschlag) [reply Mastodon/Scene/HomeTimeline/HomeTimelineViewController.swift:174]: I absolutely agree and am working on it in IOS-234 (when I'm not fixing bugs) :)

**Pull request merged**